projects/step-frontend/src/lib/modules/timeseries/time-series-context.ts

```diff
@@ -8,36 +8,62 @@ import { BucketFilters } from './model/bucket-filters';
 import { TimeSelectionState } from './time-selection.state';
 import { TSTimeRange } from './chart/model/ts-time-range';
 import { TimeSeriesUtils } from './time-series-utils';
+import { TsFilterItem } from './performance-view/filter-bar/model/ts-filter-item';
+import { TimeSeriesContextParams } from './time-series-context-params';
 
 /**
  * This class is responsible for managing the state of an execution tab. Here we store time selection, colors, filters, etc.
  */
 export class TimeSeriesContext {
-  executionId!: string;
+  id!: string;
   activeExecution: Execution | undefined;
 
+  inProgress$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
```

Comment:
Subjects in this class are not completed
Add method that will complete them and invoke it, when you're destroying the context